tests/protocols/test_protocols.py

```diff
@@ -0,0 +1,50 @@
+import tempfile
+
+import mypy.api
+import pytest
+from ophyd import sim
```

Comment:
We definitely don't want to import `ophyd` inside `ophyd_async`, even for tests. We'll have to find another way to test these. There will is an `ophyd_async.sim` that you can use, but it might not cover everything at the moment. Maybe after #57 there will be enough

Reply:
I have rebased this branch onto @gilesknap 's sim_motor branch to get a motor sim, however there doesn't appear to be any detector or flyer sim